src/components/SegmentGroupControls.vue

```diff
@@ -110,22 +124,66 @@ function createSegmentGroup() {
 
   startEditing(id);
 }
+
+function createSegmentGroupFromImage(selectedImageID: string) {
+  if (!selectedImageID)
+    throw new Error('Cannot create a labelmap without a base image');
+
+  if (primarySelection.value) {
+    segmentGroupStore.convertImageToLabelmap(
+      { type: 'image', dataID: selectedImageID },
+      primarySelection.value
+    );
+  }
+}
```

Comment:
I think this would be aggressive refactoring. While both functions look the same they're basically just checking a few things and calling `segmentGroupStore.convertImageToLabelmap()` which is already refactored.